### DISCUSSION TITLE
Encoding in H.266 (VVC) in 5.2.0?

### DISCUSSION THREAD

**Discussion by GT500org** [cdgriffith/FastFlix#425] [Q&A] [answered]
I installed FastFlix 5.2.0, I updated FFMpeg, and I restarted FastFlix however I don't see an option for x266/VVC encoding. Was support only added for decoding?

**Comment by cdgriffith** [comment, marked as answer]
Hi @GT500org, you will need to download or compile a ffmpeg build that has the `libvvenc` library in it, and link it in FastFlix. 

For example could download the one they shared in [the issue](https://github.com/cdgriffith/FastFlix/issues/233) from https://github.com/MartinEesmaa/VVCEasy/blob/master/FFMPEGVVC.md#ffmpeg-downloads-xhe-aac--vvc-endecoder-plugin-compiled-by-martin-eesmaa 

Extract those files to a directory of your choice.

And then go into FastFlix, top left click File > Settings then edit `ffmpeg` and `ffprobe` paths to point to those versions, and restart FastFlix

**Reply by GT500org** [reply]
Interesting. Hopefully support will be enabled in more commonly downloaded versions of FFmpeg for Windows soon too.

**Reply by GT500org** [reply]
Unfortunately it didn't work:
`[libvvenc @ 00000198e4fe7cc0] Unknown vvenc option: colorprim.
[libvvenc @ 00000198e4fe7cc0] Unknown vvenc option: transfer.
[libvvenc @ 00000198e4fe7cc0] Unknown vvenc option: colormatrix.
[libvvenc @ 00000198e4fe7cc0] Unknown vvenc option: chromaloc.
[gif @ 00000198e4fe7200] GIF muxer supports only a single video GIF stream.
Could not write header for output file #0 (incorrect codec parameters ?): Invalid argument
Error initializing output stream 0:0 --
Conversion failed!`

That version of FFmpeg may be missing too many important options (for instance most other encoders appear to be missing). I'll try compiling my own build of FFmpeg with VVC support, and see if that works.

**Reply by cdgriffith** [reply]
Looks like it was trying to give a `.gif` extension. I noticed that switching between encoders first time can sometimes give the wrong file extensions, so I need to look at why that is. Try switching to `Copy` and back to `VVC` and see if it is showing `.mkv` as the file ending
